x-pack/platform/plugins/shared/fields_metadata/common/hashed_cache.ts

```diff
@@ -14,7 +14,7 @@ export interface IHashedCache<KeyType, ValueType> {
   reset(): void;
 }
 
-export class HashedCache<KeyType, ValueType> {
+export class HashedCache<KeyType extends object, ValueType> {
```

Comment:
Looking the `object-hash` changes I wonder if this shouldn't be

```suggestion
export class HashedCache<KeyType extends hash.NotUndefined, ValueType> {
```
?

Reply:
Thanks! Fixed in 07025e3768940e5395d2c45dbfe962eaa64e1b58